packages/office-ui-fabric-react/src/utilities/dateMath/DateMath.ts

```diff
@@ -85,6 +85,12 @@ export function setMonth(date: Date, month: number): Date {
  * @return {boolean} True if the two dates represent the same date (regardless of time-of-day), false otherwise.
  */
 export function compareDates(date1: Date, date2: Date): boolean {
+  if (date1 == null && date2 == null) {
```

Comment:
Also, you shouldn't check for `null` explicitly, as the values could also be `undefined`.

An alternative here would be
````
if (date1 === date 2) {
    return true;
} else if (!date1 || !date2) {
    return false;
} else {
    // Compare the component values here
}